core/src/nodes/tasks/manager.rs

```diff
@@ -63,9 +63,9 @@ pub struct Manager<I, O, H, E, HE, T, C = PeerId> {
     /// Identifier for the next task to spawn.
     next_task_id: TaskId,
 
-    /// Threads pool where we spawn the nodes' tasks. If `None`, then we push tasks to the
+    /// Custom executor where we spawn the nodes' tasks. If `None`, then we push tasks to the
     /// `local_spawns` list instead.
-    threads_pool: Option<ThreadPool>,
+    executor: Option<Sp>,
```

Comment:
A pity about the rippling effect of the additional type parameter. Why not just `Option<Box<dyn Spawn>>`, since the Swarm already uses such a trait object (and thus dynamic dispatch) anyway?

Reply:
The idea is that `libp2p-core` is as zero-cost and generic as possible, while `libp2p-swarm` is a bit more opinionated.
I don't really have a strong opinion on extra template parameter vs `Box<dyn Spawn>` here though.

Reply:
I'd probably go with the more convenient option unless it proves to be a meaningful bottleneck, but I'm fine with keeping the additional type parameter, if you prefer that.

Reply:
+1 for the trait object.